app/models/package_manager/maven/maven_central.rb

```diff
@@ -16,6 +16,32 @@ def self.missing_version_remover
     PackageManager::Base::MissingVersionRemover
   end
 
+  def self.latest_version(name)
+    metadata_version = super
+
+    # When maven-metadata.xml returned an interpolation string for latest version, e.g. ${revision},
+    # attempt to scrape MavenCentral's index HTML to infer the latest version.
+    if metadata_version =~ /\$\{/
+      version_listing = get_html(MavenUrl.from_name(name, repository_base, NAME_DELIMITER).base)
```

Comment:
Is it possible to pass the `get_html` call from elsewhere? Just trying to minimize the amount of methods making calls to external third parties that are unexpected and might cause delays in the processing chain (e.g. when asking an object about an attribute I would not expect it to make a call externally).

Reply:
too late, the Maven class is already riddled with these 😆 seriously though, I can try to move this into the project() method.